README.md

```diff
@@ -82,7 +84,210 @@ Requirements:
        ```
 
 Build:
-------
+-----------------
+  Aladdin currently supports two ways to build.
+
+  1. Configure with CMake, then build with Make :
+
+       If you are familiar with CMake, or you are interested in a script
+       which can build out-of-source and offer automatical LLVM/Clang
+       installation, choose [CMake](#build-with-cmake)
+
+  2. Directly build with Makefile :
+
+       If you want to keep all the things simple, or you do not have
+       CMake, choose [Makefile](#build-with-makefile)
+
+
+Build with CMake:
+-----------------
+CMake is a configure tool which allows you to out-of-source build.
+Aladdin Requires CMake newer than 2.8.12. By default, CMake
+searches for LLVM 3.4.
+
+
+1. Set `$BOOST_ROOT`,  `TRACER_HOME`, `LLVM_HOME` to let Aladdin know where the tools and libraries are.
+
+   ```
+   export BOOST_ROOT=/your/path/to/boost
+   export TRACER_HOME=/your/path/to/LLVM-Tracer
+   export LLVM_HOME=/your/path/to/llvm
+   ```
+
+2. Use CMake to configure Aladdin and build libraries and executables
+
+   ```
+   mkdir /where/you/want/to/put/your/binary
+   cd /where/you/want/to/put/your/binary
+   cmake /your/path/to/Aladdin
+   make -j4
+   ```
+
+3. (Optional) Run tests with `ctest` command
+
+   **CAUTION : Takes a long time**
+
+   **Suggest running part of them, like `"triad"` or `"unit_test"`.**
+
+   There are 1397 tests. On Intel Xeon E5-2650 2.60GHz,
+   execution in 32 jobs takes 39 minutes,
```

Comment:
Like "make -j32". one process per test, but the process number equal or less than 32 at a time. I have changed it to "execution in 32 simultaneous processes". Is it better?